src/klein/test/test_resource.py

```diff
@@ -869,6 +869,30 @@ def handle_not_found(request, failure):
         self.assertEqual(request.processingFailed.called, False)
         self.assertEqual(request.getWrittenData(), rendered)
 
+    def test_errorHandlerReturnsResource(self):
+        """
+        Resources returned by L{handle_errors} are rendered
```

Comment:
Sure. Sorry, didn't have time to change it before you merged this PR, so created #188 for this :)